test/approve.js

```diff
@@ -0,0 +1,93 @@
+
+    yield trst.approve(ACCOUNT_TWO, approvedAmount, {from: ACCOUNT_ONE})
+    let approvedForAccountTwo = yield trst.allowance(ACCOUNT_ONE, ACCOUNT_TWO)
+    assert.equal(approvedForAccountTwo.toNumber(), approvedAmount)
+
+    yield trst.transferFrom(ACCOUNT_ONE, ACCOUNT_THREE, approvedAmount, {from: ACCOUNT_TWO})
+    let accountThreeEndingBalance = yield trst.balanceOf.call(ACCOUNT_THREE)
+    assert.equal(accountThreeEndingBalance.toNumber(), (accountThreeStartingBalance.toNumber() + approvedAmount))
+    approvedForAccountTwo = yield trst.allowance(ACCOUNT_ONE, ACCOUNT_TWO)
+    assert.equal(approvedForAccountTwo.toNumber(), 0)
+  }))
+
+  it("should not allow transferring more than an account's approved balance", co(function* () {
+    let initialAmount = 10
+    let approvedAmount = 5
+    let trst = yield utils.deployTrustcoin(OWNER, MIGRATION_MASTER)
+    yield trst.transfer(ACCOUNT_ONE, initialAmount, {from: OWNER})
+    yield trst.transfer(ACCOUNT_TWO, initialAmount, {from: OWNER})
+    yield trst.transfer(ACCOUNT_THREE, initialAmount, {from: OWNER})
+    let accountOneStartingBalance = yield trst.balanceOf.call(ACCOUNT_ONE)
+    let accountTwoStartingBalance = yield trst.balanceOf.call(ACCOUNT_TWO)
+    let accountThreeStartingBalance = yield trst.balanceOf.call(ACCOUNT_THREE)
+
+    yield trst.approve(ACCOUNT_TWO, approvedAmount, {from: ACCOUNT_ONE})
+    let approvedForAccountTwo = yield trst.allowance(ACCOUNT_ONE, ACCOUNT_TWO)
+    assert.equal(approvedForAccountTwo.toNumber(), approvedAmount)
+
+    yield trst.transferFrom(ACCOUNT_ONE, ACCOUNT_THREE, approvedAmount + 1, {from: ACCOUNT_TWO})
+    approvedForAccountTwo = yield trst.allowance(ACCOUNT_ONE, ACCOUNT_TWO)
+    assert.equal(approvedForAccountTwo.toNumber(), approvedAmount) // Assure none of the tokens moved
+
+    let accountThreeEndingBalance = yield trst.balanceOf.call(ACCOUNT_THREE)
+    assert.equal(accountThreeEndingBalance.toNumber(), accountThreeStartingBalance.toNumber())
+  }))
+
+  it("should not allow approving unless our approvee's approved balance is 0", co(function* () {
+    let initialAmount = 10
+    let approvedAmount = 5
+    let trst = yield utils.deployTrustcoin(OWNER, MIGRATION_MASTER)
+    yield trst.transfer(ACCOUNT_ONE, initialAmount, {from: OWNER})
+    yield trst.transfer(ACCOUNT_TWO, initialAmount, {from: OWNER})
+    yield trst.transfer(ACCOUNT_THREE, initialAmount, {from: OWNER})
+    let accountOneStartingBalance = yield trst.balanceOf.call(ACCOUNT_ONE)
+    let accountTwoStartingBalance = yield trst.balanceOf.call(ACCOUNT_TWO)
+    let accountThreeStartingBalance = yield trst.balanceOf.call(ACCOUNT_THREE)
+
+    yield trst.approve(ACCOUNT_TWO, approvedAmount, {from: ACCOUNT_ONE})
+    let approvedForAccountTwo = yield trst.allowance(ACCOUNT_ONE, ACCOUNT_TWO)
+    assert.equal(approvedForAccountTwo.toNumber(), approvedAmount)
+
+    yield trst.approve(ACCOUNT_TWO, approvedAmount, {from: ACCOUNT_ONE})
```

Comment:
i'm not sure what we're doing here. Are we trying to set a different approval amount? i read it as if we're trying to set the same.

Even if you change the amount, can you add a short comment describing what we're doing here? i think our future selves will be appreciative